text/_util.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+export interface SplitOptions {
+  singleDelimiter?: boolean;
+  removeSpecialCharacters?: boolean;
+}
+
+export function split(
+  input: string,
+  { singleDelimiter = false, removeSpecialCharacters }: SplitOptions = {},
+) {
+  if (removeSpecialCharacters) {
+    input = input.replaceAll(/[^a-zA-Z0-9\s-_]/g, "");
+  }
+  if (singleDelimiter) {
+    if (/\s+/.test(input)) return input.split(/\s+/).filter(Boolean);
+    if (/-+/.test(input)) return input.split(/-+/).filter(Boolean);
+    if (/_+/.test(input)) return input.split(/_+/).filter(Boolean);
+  } else {
+    if (/[\s-_]+/.test(input)) return input.split(/[\s-_]+/).filter(Boolean);
+  }
+  return input.split(/(?=[A-Z])+/).filter(Boolean);
+}
```

Comment:
Let's revert this to the original version. It's good for the first-pass.

Reply:
Are you sure? I think it is an improvement that handles a bunch of edge cases concerning delimiters, whitespaces and special characters.

Reply:
Yes, for now. I know it won't be what's published upon release, but it allows us to explore ideas. Frankly, I'm not a fan of the `SplitOptions` argument here. My idea was to perhaps have a 2nd optional parameter, say `separator: string | RegExp`, that the user can define but has a reasonable default. Again, I'd want to play with the idea a little first.

Reply:
The problem with this function is that there's a seemingly arbitrary order of precedence in the characters that the word is split by. I.e. the word is split by spaces if spaces are what's first detected. Then dashes. Then underscores. Why in that order?

I think we should just split by all of these delimiters in one hit, as I previously suggested. I understand that it'll also split words with delimiters of mixed types. But that seems like the correct behaviour to me and is behaviour that's easier to understand. Also, that's what lodash does.
```ts
import camelCase from "npm:lodash.camelcase";
console.log(camelCase("why hello-there")); // "whyHelloThere"
```